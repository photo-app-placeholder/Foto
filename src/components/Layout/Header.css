header {
  display: flex;
  justify-content: space-between;
  margin: 20px;
}

header ul {
  display: flex;
  flex-direction: row;
  list-style-type: none;
  margin: 0;
}

header li {
  padding: 10px 20px 10px 20px;
  width: 70px;
  margin: 5px;
  border: 1pt solid gray;
  border-radius: 10px;
}

header a {
  text-decoration: none;
}

.home {
  padding: 10px 20px 10px 20px;
  width: 70px;
  margin: 5px;
}

header img {
  height: 35px;
  margin-right: 30px;
  margin-top: 5px;
}
header button {
  width: fit-content;
  height: 20px;
  margin: 20px 15px 10px 0px;
}

.usernameText {
  margin: 20px 30px 0px 0px;
}
