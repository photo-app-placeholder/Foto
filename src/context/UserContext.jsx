import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

export const UserContext = createContext();

export const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  console.log(user);

  const contextValue = { user, setUser };

  return (
    <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>
  );
};

export const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('Error: useUser needs to be used within UserContext');
  }
  return context;
};
