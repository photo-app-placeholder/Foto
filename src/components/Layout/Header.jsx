import React from 'react';
import { Link } from 'react-router-dom';
import { useUser } from '../../context/UserContext';
import { signOutUser } from '../../services/users';
import profileHook from '../../hooks/profileHook';
import profilePic from '../../assets/profile.png';
import styles from './Header.css';
const { home, usernameText } = styles;

export default function Header() {
  const { user, setUser } = useUser();
  const { profile } = profileHook();
  const { username } = profile[0] || '';

  const handleLogout = async () => {
    await signOutUser();
    setUser({});
  };
  return (
    <header>
      <div className={home}>
        <Link to="/">PhotoApp</Link>
      </div>
      <div>
        {!user.email && (
          <>
            <ul>
              <Link to="/login">
                <li>Login</li>
              </Link>
              <Link to="/register">
                <li>Register</li>
              </Link>
            </ul>
          </>
        )}
        {user.email && (
          <>
            <ul>
              <h4 className={usernameText}>@{username}</h4>
              <Link to={`/${username}`}>
                <img src={profilePic} />
              </Link>
              <button onClick={handleLogout}>Log Out</button>
            </ul>
          </>
        )}
      </div>
    </header>
  );
}
