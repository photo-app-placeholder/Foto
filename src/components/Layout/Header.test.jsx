import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import Header from './Header';

test('should display header', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
