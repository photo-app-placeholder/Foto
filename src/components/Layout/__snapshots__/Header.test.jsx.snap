// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display header 1`] = `
<div>
  <header>
    <h1
      class="home"
    >
      <a
        href="/"
      >
        PhotoApp
      </a>
    </h1>
    <div>
      <ul>
        <a
          href="/login"
        >
          <li>
            Login
          </li>
        </a>
        <a
          href="/register"
        >
          <li>
            Register
          </li>
        </a>
      </ul>
    </div>
  </header>
</div>
`;
